tests/unit/unit_tests_common.c

```diff
@@ -12,71 +12,30 @@
- * when the caller sets the VERBOSE env var to opt-in.
- *
- * NOTE about exp_msg_level in set_log_streams_for_*tests() functions:
- *
- * A unit-test, or a test case, that invokes these interfaces is declaring that
- * messages logged when running this test/case will be visible only when the
- * message verbosity is >= than this expected message level. Therefore, to see
- * these messages, the test-runner will have to invoke the test/case with a
- * verbosity level >= expected message level. Otherwise, the output log handles
- * will be set to /dev/null which will suppress all test output.
- */
-void
-set_log_streams_for_tests(msg_level exp_msg_level)
-{
-   if (Ctest_verbosity >= exp_msg_level) {
-      platform_set_log_streams(stdout, stderr);
-   } else {
-      platform_reset_log_streams();
-   }
-}
-
-/*
- * Setup function is provided to manage output log streams from (unit) tests
- * that are induce error conditions and check that error messages are raised
- * as expected.
+ * Setup function is provided to manage output log streams from (unit) tests.
+ * By default, execution from unit-tests will be silent, redirecting output
+ * handles to /dev/null. Info / error messages are only printed when the caller
+ * sets the VERBOSE env var to opt-in.
  *
  * Some unit tests exercise error cases, so even when everything succeeds, they
  * generate lots of "error" messages. By default, that would go to stderr, which
  * would pollute the test output. This function sets up output file handles such
  * that those expected error messages are only printed when the caller sets the
- * VERBOSE env var to opt-in. By default, execution from unit-tests will be
- * silent, redirecting output handles to /dev/null.
- *
- * Returns: Output file handles (out/error) if caller has requested for them.
+ * VERBOSE env var to opt-in.
  */
 void
-set_log_streams_for_error_tests(msg_level             exp_msg_level,
-                                platform_log_handle **log_stdout,
-                                platform_log_handle **log_stderr)
+set_log_streams_for_tests(msg_level exp_msg_level)
 {
    // Here we ensure those expected error messages are only printed
    // when the caller sets the VERBOSE env var to opt-in.
    if (Ctest_verbosity >= exp_msg_level) {
       platform_set_log_streams(stdout, stderr);
-      if (log_stdout) {
-         *log_stdout = stdout;
-      }
-      if (log_stderr) {
-         *log_stderr = stderr;
-      }
-      CTEST_LOG_INFO("\nVerbose mode on. This test may exercise an error case, "
-                     "so on success it will print a message that appears to "
-                     "be an error. "
-                     "Or, the test may exercise print-diagnostic code.\n");
+      CTEST_LOG("Verbose mode on. This test may exercise an error case, so on"
+                " success it may print a message that appears to be an error."
+                " Or, the test may exercise print-diagnostic code.\n");
```

Comment:
One thing I didn't consider when we discussed this before is that, now that we've combined these two functions together, if you run a non-error-exercising test with `VERBOSE=3` you'll see this message, which is kind of noise since it is never true.  

I think I'd be ok with either of the following "fixes":
1. Just shorten this message so it's less than 80 characters, so at least its not too much noise
2. Add a boolean argument to this function, something like `bool may_print_errors`, which would control whether or not this log line is shown.  And then pass in `true` from `limitations_test` and such, but `false` otherwise.

Reply:
@rosenhouse - I did a large'ish rework of all this handling and updated `set_log_streams_for_tests()` to a large extent. Lots has changed.

See [this commit](https://github.com/vmware/splinterdb/pull/541/commits/00d9c5abc0e29fdc287b580f91a33e0ea14e34f5).